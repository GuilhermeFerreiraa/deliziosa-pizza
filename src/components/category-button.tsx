import { clsx } from 'clsx';
import { Pressable, PressableProps, Text } from 'react-native';

type CategoryProps = PressableProps & {
 title: string;
 isSelected?: boolean;
}

export default function CategoryButton({ title, isSelected, ...rest }: CategoryProps) {
 return (
  <Pressable className={clsx('bg-primary px-4 justify-center rounded-lg h-10', isSelected && 'border-2 border-primary bg-slate-100')} {...rest}>
   <Text className={clsx('text-slate-100 font-subtitle text-sm', isSelected && 'text-primary font-heading')}>
    {title}
   </Text>
  </Pressable>
 )
}