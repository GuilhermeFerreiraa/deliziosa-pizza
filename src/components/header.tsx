import { Image, Text, View, TouchableOpacity, Pressable } from "react-native";
import { Feather } from "@expo/vector-icons";
import colors from "tailwindcss/colors";
import { Link } from "expo-router";

type HeaderProps = {
  title: string;
  cartQuantity?: number;
};

export default function Header({ title, cartQuantity = 0 }: HeaderProps) {
  return (
    <View className="flex items-center border-b border-slate-700 pb-5 mx-5">
      <View className="w-full flex-row items-center justify-between">
        <Image source={require("@/assets/logo_black.png")} className="h-14 w-14" />
        <Text className="text-gray-800 text-left text-xl font-heading">Bem-vindo(a)!</Text>
        <View className="flex-row gap-x-4 items-center">
          {cartQuantity > 0 && (
            <Link href="/cart" asChild>
              <TouchableOpacity className="relative mb-4" activeOpacity={0.7}>
                <View className="bg-secondary w-4 h-4 rounded-full items-center justify-center top-2 z-10 -right-3.5">
                  <Text className="text-slate-900 font-bol text-xs">
                    {cartQuantity}
                  </Text>
                </View>
                <Feather name="shopping-bag" color={colors.slate[800]} size={24} />
              </TouchableOpacity>
            </Link>
          )}
          <Link href="/" asChild>
            <Pressable>
              <Feather name="log-out" color={colors.slate[800]} size={24} />
            </Pressable>
          </Link>
        </View>
      </View>
      <Text className="text-slate-800 w-full text-left mt-2 text-xl font-heading">{title}</Text>
    </View>
  );
}
