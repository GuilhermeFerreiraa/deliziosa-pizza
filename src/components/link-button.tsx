import { Link } from "expo-router";
import { ExpoRouter } from "../../.expo/types/router";
import { Pressable, Text } from "react-native";

type LinkButtonProps = ExpoRouter.LinkProps<string> & {
  title: string;
};

export function LinkButton({ title, ...rest }: LinkButtonProps) {
  return (
    <Link asChild {...rest}>
      <Pressable>
        <Text className="text-slate-800 text-center text-base font-body">
          {title}
        </Text>
      </Pressable>
    </Link>
  );
}
