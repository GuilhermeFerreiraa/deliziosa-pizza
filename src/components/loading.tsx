import { ActivityIndicator, View } from "react-native";
import colors from 'tailwindcss/colors'

export default function Loading() {
 return (
  <View className="flex-1 absolute w-full h-full opacity-75 z-50 items-center justify-center bg-primary">
   <ActivityIndicator color={colors.white} size={24}/>
  </View>
 )
}