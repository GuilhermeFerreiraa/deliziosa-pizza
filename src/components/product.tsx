import {
  Image,
  ImageProps,
  Text,
  TouchableOpacity,
  TouchableOpacityProps,
  View,
} from "react-native";
import { forwardRef } from "react";
import { formatCurrency } from "@/utils/functions/format-currency";

type ProductDataProps = {
  title: string;
  description: string;
  thumbnail: ImageProps;
  quantity?: number;
  price?: number;
};

type ProductProps = TouchableOpacityProps & {
  data: ProductDataProps;
};

const Product = forwardRef<TouchableOpacity, ProductProps>(
  ({ data, ...rest }, ref) => {
    return (
      <TouchableOpacity
        ref={ref}
        className="w-full flex-row items-center pb-4 bg-white p-2 rounded-md mb-2"
        {...rest}
      >
        <Image source={data.thumbnail} className="w-20 h-20 rounded-md" />
        <View className="flex-1 ml-3">
          <View className="flex-row items-center">
            <Text numberOfLines={1} className="text-slate-900 w-full flex-1 font-subtitle text-base">
              {data.title}
            </Text>
             <Text className="text-slate-800 font-subtitle text-xs">
              {data.price ? formatCurrency(data.price) : 'R$0,00'}
            </Text>
          </View>
            {data.quantity && (
              <Text className="text-slate-800 font-subtitle text-sm">
                x{data.quantity}
              </Text>
            )}

          <Text className="text-slate-800 text-xs leading-5 mt-0.5">
            {data.description}
          </Text>
        </View>
      </TouchableOpacity>
    );
  }
);

export default Product;
