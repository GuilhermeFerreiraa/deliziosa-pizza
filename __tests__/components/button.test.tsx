import { Button } from "@/components/button";
import React from "react";
import renderer from "react-test-renderer";

describe("Button Component", () => {
  it("renders children correctly", () => {
    const component = renderer.create(
      <Button>
        <Button.Text>Click Me</Button.Text>
      </Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
