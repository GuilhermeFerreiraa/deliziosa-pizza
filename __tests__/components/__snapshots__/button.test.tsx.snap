// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders children correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#BF0A25",
      "borderBottomLeftRadius": 6,
      "borderBottomRightRadius": 6,
      "borderTopLeftRadius": 6,
      "borderTopRightRadius": 6,
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
        },
        {
          "fontFamily": "Inter_600SemiBold",
        },
        {
          "textAlign": "center",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "marginLeft": 8,
          "marginRight": 8,
        },
      ]
    }
  >
    Click Me
  </Text>
</View>
`;
